File: examples/tentacles/polyglot/echo.pl
use strict;
use JSON::PP;
my $raw = do { local $/; <STDIN> } // '{}';
$raw = '{}' unless length $raw;
my $req = decode_json($raw);
print encode_json({
  v => 1, id => $req->{id}, ok => JSON::PP::true,
  result => { arm => "echo-pl", lang => "perl", action => $req->{action}, echo => $req->{payload} // {} }
});

File: examples/tentacles/polyglot/echo.rb
require "json"
raw = STDIN.read
raw = "{}" if raw.nil? || raw.empty?
req = JSON.parse(raw)
print JSON.generate({
  "v" => 1, "id" => req["id"], "ok" => true,
  "result" => { "arm" => "echo-rb", "lang" => "ruby", "action" => req["action"], "echo" => req["payload"] || {} }
})
